### PR TITLE
Create entities

### PR DESCRIPTION
**Pull request by riteshsp2000**
### Your checklist for this pull request

🚨Please review the [guidelines for contributing](../CONTRIBUTING.md) to this
repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch**
      (right side). Don't request your default branch!
- [x] Make sure you are making a pull request against the **default branch**
      (left side). Also you should start _your branch_ off _default branch_.
- [x] Check the commit's or even all commits' message styles matches our
      requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit
      test.
- [x] I have added necessary documentation (if appropriate)

### Description
Create TypeGraphQL object type for the following:
1. User 
2. Quiz
3. Question
- Mongoose Model, Typescript Interface and GraphQL Object type for all are created using a single TypeGraphQL Object type decorator.

Please describe your pull request.

❤️Thank you!

### Post merge checklist

- [ ] Follow steps from the [guidelines for contributing](../CONTRIBUTING.md) to
      this repository.
- [ ] If you are a new contributor, ping in the thread and one of the
      maintainers will add you to all-contributors list.

### CODE REVIEW

File: src/schema/user/user.model.ts [hunk 1]
```diff
@@ -1,27 +1,74 @@
...
-  @Property({default: new Date()})
-  @GqlField(() => Date)
-  createdAt: Date;
+  @Property({
+    required: true,
+    unique: true,
+    trim: true,
+    lowercase: true,
+    match:
+      /^(([^<>()[\].,;:\s@"]+(\.[^<>()[\].,;:\s@"]+)*)|(".+"))@gmail(.([^<>()[\].,;:\s@"]{2,})){1,}$/i,
```
Comment (abhibhaw) [inline src/schema/user/user.model.ts:29]: @riteshsp2000 is only gmail allowed for user registration?

File: src/schema/quiz/quiz.model.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    description: 'A list of all the instructions for the quiz',
+  })
+  instructions: string[];
+
+  @Property({default: []})
+  @Field(() => [String], {
+    description:
+      'An array of objects containing the student id and corresponding marks',
+  })
+  submissions: {id: string; marks: number}[];
```
Comment (abhibhaw) [inline src/schema/quiz/quiz.model.ts:64]: We are storing students I'd for a single quiz twice, once here and other in user model for quiz id's array

File: src/schema/user/user.model.ts (hunk 1 above)
Comment (riteshsp2000) [reply src/schema/user/user.model.ts:29]: are we planning to allow any others? if so ill update if not then we can keep it the same. And also I had google oauth in mind.

File: src/schema/quiz/quiz.model.ts (hunk 2 above)
Comment (riteshsp2000) [reply src/schema/quiz/quiz.model.ts:64]: Yes agreed.
- In the user model, we are storing the list of quiz ids the user has attempted.
- Here we are saving the users' IDs who submitted the quiz, along with that we are also storing its corresponding marks. If we remove the IDs there is no relation between the marks and IDs stored in the DB.

**Review by abhibhaw — APPROVED** [review]
🎉🎉

**Pull request merged**